include/snitch/snitch_registry.hpp

```diff
@@ -330,6 +337,150 @@ class registry {
     SNITCH_EXPORT small_vector_span<const registered_reporter> reporters() const noexcept;
 };
 
+#else // SNITCH_DISABLE
+
+class registry {
```

Comment:
This is a lot of code duplication, it will be painful to maintain. Maybe we can find a better solution. What are you trying to achieve? If you want to avoid the big `test_list` array, then we could force all `max_*` variables (including `max_test_cases`) to `0` (if possible, `1` otherwise) when *snitch* is disabled, and leave the whole `registry` class the same whether *snitch* is enabled or disabled. 

But maybe an even simpler solution is to simply not declare the global registry when *snitch* is disabled:
```
SNITCH_EXPORT extern constinit registry tests;
```

This should be unused when *snitch* is disabled, so should be safe to not declare/define at all.

Reply:
Globally speaking, the reason of the code duplication is to make the compiler minimize the binary size of the executable (essentially its text section), also when the program is compiled in debug mode (without any optimization).

The global snitch::tests variable has not been deliberately removed in order to preserve the build of the test cases (when SNITCH_DISABLE is ON) and, in general, of any program that makes use of the snitch::tests.run_tests(args) API.

By doing the above, we managed to reach the maximum code size reduction with GCC (when LTO is enabled), where the only remaining symbol is the global registry (i.e. occupying 44 bytes on x86 64 bit).